asn1/CommonLedgerProtocol.asn

```diff
@@ -0,0 +1,114 @@
+CommonLedgerProtocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Timestamp
+    FROM InterledgerTypes
+
+    InterledgerPacket
+    FROM InterledgerPacket
+;
+
+-- Zero or more protocolData entries, each with a unique
+-- protocolName, describe protocol-specific information about
+-- unpaid and conditionally paid requests and results
+-- sent over the CommonLedgerProtocol.
+-- The protocolName of each entry describes the protocol for
+-- interpreting the data. Peers need to establish beforehand
+-- which protocols each understand, and which names they
+-- use, which can be agreed ad-hoc, except that protocolName 'ilp'
+-- is reserved for the Interledger protocol.
+-- When multiple protocolData items are sent in a request,
+-- only the first one will be decisive in triggering the
+-- response, the rest of the protocolData items are considered
+-- side protocols that piggyback on the first one.
+-- The contentType byte can be used to allow debug tools to
+-- display the contents of the protocolData without understanding
+-- it.
+
+ContentType ::= INTEGER {
```

Comment:
This field is intended to give applications that don't know/understand the custom protocol a way to at least partially decode the data. For example. we have values to indicate whether it's binary or text data. We intentionally want a small, defined list of supported formats here, NOT support any arbitrary mime type. If we want to add a format to the list, there should be broad agreement in the Interledger community that's it is worth adding this format to a large number of implementations. If the values in this field aren't widely understood the field is not useful.

So I don't think we redefining anything - we're just defining a short list of mime types that are widely understood and since it's a small, standardized list, an enum is the correct type to use. I just realized that there is an `ENUMERATED` type in ASN.1 which would perhaps be an even better fit than `INTEGER`.